app/models/concerns/status/fasp_concern.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Status::FaspConcern
+  extend ActiveSupport::Concern
+
+  included do
+    after_commit :announce_new_content_to_subscribed_fasp, on: :create
+    after_commit :announce_updated_content_to_subscribed_fasp, on: :update
+    after_commit :announce_deleted_content_to_subscribed_fasp, on: :destroy
+    after_commit :announce_trends_to_subscribed_fasp, on: :create
+  end
+
+  private
+
+  def announce_new_content_to_subscribed_fasp
+    return unless Mastodon::Feature.fasp_enabled?
+    return unless account_indexable? && public_visibility?
+
+    # We need the uri here, but it is set in another `after_commit`
+    # callback. Hooks included from modules are run before the ones
+    # in the class itself and can neither be reordered nor is there
+    # a way to declare dependencies.
+    store_uri if uri.nil?
```

Comment:
An alternative, I think, is to do `prepend: true` in the `after_commit` in the main class. Not sure this really is better though.

Reply:
That looks like to be the case indeed, good catch. This is a bit annoying.